pandas/tests/extension/base/getitem.py

```diff
@@ -120,6 +120,33 @@ def test_getitem_scalar(self, data):
         result = pd.Series(data)[0]
         assert isinstance(result, data.dtype.type)
 
+    def test_getitem_invalid(self, data):
+        # TODO: box over scalar, [scalar], (scalar,)?
+
+        msg = (
+            r"only integers, slices \(`:`\), ellipsis \(`...`\), numpy.newaxis "
+            r"\(`None`\) and integer or boolean arrays are valid indices"
+        )
+        with pytest.raises(IndexError, match=msg):
+            data["foo"]
+        with pytest.raises(IndexError, match=msg):
+            data[2.5]
+
+        ub = len(data)
+        msg = "|".join(
+            [
+                "list index out of range",  # json
+                "index out of bounds",  # pyarrow
+                "Out of bounds access",  # Sparse
+                f"index {ub+1} is out of bounds for axis 0 with size {ub}",
+                f"index -{ub+1} is out of bounds for axis 0 with size {ub}",
```

Comment:
> >       with pytest.raises(IndexError, match=msg):
E       AssertionError: Regex pattern 'list index out of range|index out of bounds|Out of bounds access|index 101 is out of bounds for axis 0 with size 100|index -101 is out of bounds for axis 0 with size 100' does not match 'loc must be an integer between -100 and 100'.

i think these are asserting on 3.10 builds

Reply:
https://github.com/pandas-dev/pandas/runs/4058191626?check_suite_focus=true

Reply:
should be addressed by just-pushed commit in #44242